prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,334 @@
+use proc_macro2::{TokenStream, TokenTree};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+
+// generate a hash of path
+fn hash_path(path: syn::Path) -> u64 {
+    use std::collections::hash_map::DefaultHasher;
+    use std::hash::{Hash, Hasher};
+    let mut hasher = DefaultHasher::new();
+    path.hash(&mut hasher);
+    hasher.finish()
+}
+
+// generate a macro identifier with given path in this format:
+// {last_segment in path}_{hash(path)}
+pub fn generate_macro_ident(path: syn::Path) -> syn::Ident {
+    let ident = path.segments.last().unwrap().ident.to_owned();
+    syn::Ident::new(&format!("{}_{}_{}", "prusti_export", ident, hash_path(path)), ident.span())
+}
+
+// generate macro for export based on given identifier and item
+fn generate_macro_for_export<T: ToTokens>(
+    macro_ident: syn::Ident,
+    macro_content: T,
+) -> TokenStream {
+    quote_spanned! {macro_ident.span() =>
+        #[macro_export]
+        #[prusti::export_spec_macro]
+        macro_rules! #macro_ident {
+            () => {
+                #macro_content
+            };
+        }
+    }
+}
+
+// helper macro for getting the value of attribute with given name in an item
+// attribute must be in this format: #[name(value)]
+// if attribute is not present then path is not changed
+// @param: $path_mut_ref must be a mutable reference of Path, i.e. &mut syn::Path / mut syn::Path
+macro_rules! get_prusti_path_in_attr {
```

Comment:
This is extremely ugly… Please just make this a function that returns a value, rather than accepting an argument that is filled in.